### PR TITLE
#14353: DRAM Read Alignment for Layernorm

### PR DESCRIPTION
**Pull request by ncvetkovicTT**
### Ticket
#14353 #14354 #14350

### Problem description
Blackhole has a 64B-aligned DRAM access, Wormhole has 32B alignment. Inside four LayerNorm / Softmax data movement kernels, gamma and beta parameters are read into L1 assuming 32B DRAM alignment.

### What's changed
The four affected kernels are adjusted for 64B DRAM alignment. Tests in ```test_layernorm_sharded.py```, ```test_distributed_layernorm_post_allgather.py``` and ```test_softmax_sharded.py``` are uncommented.

### Checklist
- [x] Post commit CI passes - [#22341](https://github.com/tenstorrent/tt-metal/actions/runs/12316972401)
- [x] Blackhole Post commit (if applicable) - [Fails the same way as on main](https://github.com/tenstorrent/tt-metal/actions/runs/12312945601)
- [ ] Model regression CI testing passes (if applicable)
- [ ] Device performance regression CI testing passes (if applicable)
- [ ] **(For models and ops writers)** Full [new models](https://github.com/tenstorrent/tt-metal/actions/workflows/full-new-models-suite.yaml) tests passes
- [x] New/Existing tests provide coverage for changes

### CODE REVIEW

**Review by bbradelTT — APPROVED** [review]

File: ttnn/cpp/ttnn/operations/normalization/softmax/device/kernels/dataflow/reader_unary_sharded_sm_rm_mask.cpp [hunk 1]
```diff
@@ -42,8 +42,9 @@ void kernel_main() {
     uint32_t l1_write_addr = get_write_ptr(cb_attn);
     for (uint32_t w = 0; w < block_wt; w++) {
         uint64_t mask_noc_addr = get_noc_addr(mask_start_tile_id + w, addr_mask);
-        noc_async_read(mask_noc_addr, l1_write_addr, mask_read_tile_face_bytes);
-        mask_noc_addr += mask_read_tile_face_bytes;
+        noc_async_read(mask_noc_addr, l1_write_addr, mask_read_tile_face_bytes * 2);
```
Comment (tt-aho) [inline ttnn/cpp/ttnn/operations/normalization/softmax/device/kernels/dataflow/reader_unary_sharded_sm_rm_mask.cpp:45]: Minor: Could you make the declaration of `mask_read_tile_face_bytes` and `mask_read_tile_offset_bytes` constexpr?

File: ttnn/cpp/ttnn/operations/normalization/layernorm/device/kernels/dataflow/writer_unary_sharded_ln_rm_gb.cpp [hunk 2]
```diff
@@ -67,8 +67,9 @@ void kernel_main() {
         for (uint32_t w = 0; w < block_w; w++) {
             uint32_t tile_id = gamma_tile_start_id + w;
             uint64_t gamma_noc_addr = get_noc_addr(tile_id, gamma);
-            noc_async_read(gamma_noc_addr, l1_write_addr_gamma, mask_read_tile_face_bytes);
-            gamma_noc_addr += mask_read_tile_face_bytes;
+            noc_async_read(gamma_noc_addr, l1_write_addr_gamma, mask_read_tile_face_bytes * 2);
```
Comment (tt-aho) [inline ttnn/cpp/ttnn/operations/normalization/layernorm/device/kernels/dataflow/writer_unary_sharded_ln_rm_gb.cpp:70]: Minor: Could you make the declaration of `mask_read_tile_face_bytes` and `mask_read_tile_offset_bytes` constexpr?

**Review by tt-aho — APPROVED** [review]
Looks fine if perf pipeline was unaffected by these changes.

**Comment by tt-aho** [comment]
This is the single device perf pipeline: https://github.com/tenstorrent/tt-metal/actions/workflows/perf-models.yaml

We have perf pipelines for t3k, tg, tgg as well. Maybe run t3k as well https://github.com/tenstorrent/tt-metal/actions/workflows/t3000-model-perf-tests.yaml

**Comment by ncvetkovicTT** [comment]
@tt-aho Help, how do I interpret this failure? https://github.com/tenstorrent/tt-metal/actions/runs/12317695443

**Comment by tt-aho** [comment]
> @tt-aho Help, how do I interpret this failure? https://github.com/tenstorrent/tt-metal/actions/runs/12317695443
> 
> Edit: waiting for others [Device perf regression](https://github.com/tenstorrent/tt-metal/actions/runs/12317765324), [Model perf tests](https://github.com/tenstorrent/tt-metal/actions/runs/12317697514)

This means that the model perf was slower than expected, normally this would indicate a regression however it looks like that test has highly variable perf, ex this recent run from main also failed. https://github.com/tenstorrent/tt-metal/actions/runs/12307176953/job/34350488927
You can try rerunning that llama job again to see if it's consistently worse or not, but it's probably okay to ignore since it also sometimes fails on main.

**Comment by ncvetkovicTT** [comment]
[Device perf regression passed](https://github.com/tenstorrent/tt-metal/actions/runs/12317765324)[
T3000 model perf tests](https://github.com/tenstorrent/tt-metal/actions/runs/12317695443)
[Model perf tests](https://github.com/tenstorrent/tt-metal/actions/runs/12317697514)

I guess this is good to be merged, what do you thing @tt-aho @bbradelTT ?

**Pull request merged**